src/Protocol/ActivityPub/Processor.php

```diff
@@ -1347,18 +1346,22 @@ public static function storeReceivers(int $uriid, array $receivers)
 	/**
 	 * Creates an mail post
 	 *
-	 * @param array $activity Activity data
-	 * @param array $item     item array
+	 * @param array $item item array
 	 * @return int|bool New mail table row id or false on error
 	 * @throws \Friendica\Network\HTTPException\InternalServerErrorException
 	 */
-	private static function postMail(array $activity, array $item)
+	private static function postMail(array $item): bool
 	{
 		if (($item['gravity'] != Item::GRAVITY_PARENT) && !DBA::exists('mail', ['uri' => $item['thr-parent'], 'uid' => $item['uid']])) {
 			Logger::info('Parent not found, mail will be discarded.', ['uid' => $item['uid'], 'uri' => $item['thr-parent']]);
 			return false;
 		}
 
+		if (!Contact::isFollower($item['contact-id'], $item['uid']) && !Contact::isSharing($item['contact-id'], $item['uid'])) {
```

Comment:
What about the `unkmail` setting where users can accept private messages even by people they don't follow or are followed by?

Reply:
You cannot directly reply to messages you received that way. That setting is completely outdated.

Reply:
Hmm.